DAOS-11247 build: set memlock to unlimited

Michael-Hennecke opened the pull request:
add /etc/security/limits.d/daos_limits.conf on DAOS servers to set memlock limit to unlimited (for non-systemd environments).

Signed-off-by: Michael Hennecke <michael.hennecke@intel.com>

Michael-Hennecke committed:
DAOS-11247 build: set memlock to unlimited

add /etc/security/limits.d/daos_limits.conf on DAOS servers
to set memlock limit to unlimited /for non-systemd environments).

Signed-off-by: Michael Hennecke <michael.hennecke@intel.com>

daosbuild1 reviewed (commented):
LGTM.  No errors found by checkpatch.

Michael-Hennecke commented on utils/rpms/daos.spec:
```diff
@@ -360,6 +361,7 @@ mkdir -p %{buildroot}/%{_sysconfdir}/ld.so.conf.d/
 echo "%{_libdir}/daos_srv" > %{buildroot}/%{_sysconfdir}/ld.so.conf.d/daos.conf
 mkdir -p %{buildroot}/%{_sysctldir}
 install -m 644 utils/rpms/%{sysctl_script_name} %{buildroot}/%{_sysctldir}
+#install -m 644 utils/rpms/%{limits_script_name} %{buildroot}/%{_limitsdir}
```
@brianjmurrell could you advise how to properly specify the _limitsdir ?

brianjmurrell replied:
I'm not aware of a pre-defined `%{_limitsdir}` on either of EL8 or Leap 15.3:

```
$ rpm --eval %{_limitsdir}
%{_limitsdir}
```

```suggestion
install -m 644 utils/rpms/%{limits_script_name} %{buildroot}/%{_sysconfdir}/security/limits.d/
```

brianjmurrell commented on utils/rpms/daos.spec:
```diff
@@ -453,6 +455,7 @@ getent passwd daos_agent >/dev/null || useradd -s /sbin/nologin -r -g daos_agent
 %exclude %{_datadir}/%{name}/ioil-ld-opts
 %{_unitdir}/%{server_svc_name}
 %{_sysctldir}/%{sysctl_script_name}
+#%{_limitsdir}/%{limits_script_name}
```
```suggestion
#%{_sysconfdir}/security/limits.d/%{limits_script_name}
```

Michael-Hennecke committed:
DAOS-11247 build: set memlock to unlimited

fix install path in daos.spec

Signed-off-by: Michael Hennecke <michael.hennecke@intel.com>

daosbuild1 reviewed (commented):
LGTM.  No errors found by checkpatch.

brianjmurrell commented on utils/rpms/daos.spec:
```diff
@@ -360,6 +361,7 @@ mkdir -p %{buildroot}/%{_sysconfdir}/ld.so.conf.d/
 echo "%{_libdir}/daos_srv" > %{buildroot}/%{_sysconfdir}/ld.so.conf.d/daos.conf
 mkdir -p %{buildroot}/%{_sysctldir}
 install -m 644 utils/rpms/%{sysctl_script_name} %{buildroot}/%{_sysctldir}
+install -m 644 utils/rpms/%{limits_script_name} %{buildroot}/%{_sysconfigdir}/security/limits.d
```
I should have noticed this before, but you need a `mkdir` here like you did for debian: https://github.com/daos-stack/daos/blob/fa0f5e8bdfa1ca42b248f0eb65e8b05ca3e0795b/debian/rules#L58

```suggestion
mkdir -p ${buildroot}$(sysconfdir)/security/limits.d
install -m 644 utils/rpms/%{limits_script_name} %{buildroot}/%{_sysconfigdir}/security/limits.d
```

daosbuild1 commented:
Test stage Build RPM on EL 8 completed with status FAILURE.  https://build.hpdd.intel.com//job/daos-stack/job/daos/view/change-requests/job/PR-10262/2/execution/node/317/log

daosbuild1 commented:
Test stage Build DEB on Ubuntu 20.04 completed with status FAILURE.  https://build.hpdd.intel.com//job/daos-stack/job/daos/view/change-requests/job/PR-10262/2/execution/node/321/log

daosbuild1 commented:
Test stage Build RPM on Leap 15 completed with status FAILURE.  https://build.hpdd.intel.com//job/daos-stack/job/daos/view/change-requests/job/PR-10262/2/execution/node/368/log

brianjmurrell commented on utils/rpms/daos_limits.conf:
```diff
@@ -0,0 +1,16 @@
+#
+# <domain>        <type>  <item>  <value>
+
+* soft memlock unlimited
+* hard memlock unlimited
```
It's slightly concerning that we are raising this limit for every{one,thing} on a system where this is installed rather the much more fine-grained limit increase that is done for just the daos processes by the systemd unit file.

I'd like to [understand the use-case](https://daosio.atlassian.net/browse/DAOS-11247?focusedCommentId=104170) for this a bit better.

Michael-Hennecke replied:
see my update in https://daosio.atlassian.net/browse/DAOS-11247

brianjmurrell reviewed (changes requested):
I did [ask about the particular use-case for this in the ticket](https://daosio.atlassian.net/browse/DAOS-11247?focusedCommentId=104170), if you could find a few minutes to respond just so that I understand why this is necessary.  P&T.

johannlombardi reviewed (changes requested):
Patch has a merge conflict now.

Michael-Hennecke committed:
DAOS-11247 build: set memlock to unlimited

Merge branch 'master' into michael-hennecke/2.3/daos-11247-memlock
Update DAOS version

Signed-off-by: Michael Hennecke <michael.hennecke@intel.com>

Michael-Hennecke commented:
merged master and updated the DAOS version info

daosbuild1 reviewed (commented):
LGTM.  No errors found by checkpatch.

daosbuild1 commented:
Test stage Build RPM on EL 8 completed with status FAILURE.  https://build.hpdd.intel.com//job/daos-stack/job/daos/view/change-requests/job/PR-10262/3/execution/node/367/log

daosbuild1 commented:
Test stage Build DEB on Ubuntu 20.04 completed with status FAILURE.  https://build.hpdd.intel.com//job/daos-stack/job/daos/view/change-requests/job/PR-10262/3/execution/node/323/log
